src/components/application_manager/src/app_launch/app_launch_ctrl_impl.cc

```diff
@@ -83,6 +83,8 @@ bool HmiLevelSorter(const std::pair<int32_t, ApplicationDataPtr>& lval,
 
 void AppLaunchCtrlImpl::OnDeviceConnected(const std::string& device_mac) {
   SDL_LOG_AUTO_TRACE();
+  sync_primitives::AutoLock lock(launch_ctrl_lock_);
```

Comment:
As I understand, this lock was used to synchronize "DeviceAppsLauncher device_apps_launcher_;" calls whish are not present in OnAppRegistered function.

Reply:
hmmm, in that case perhaps the scope of the lock should be reduced in this function? I was under the impression that this covered multiple member variables because of how vague the name was.

It should also probably be renamed to something like `device_apps_launcher_lock_` accordingly

Reply:
Done